SC_redesign/Crop_and_Soil/soil/soil_data.py

```diff
@@ -30,16 +31,44 @@ class SoilData:
     previous_retention_parameter: Optional[float] = None
     """retention parameter for the previous day (mm) (used in SWAT 2:1.1.9)"""
     average_slope_fraction: float = 0.05
-    """average slope fraction of the subbasin"""
+    """average slope fraction of the subbasin (unitless)"""
     moisture_condition_parameter: Optional[float] = None
     """curve number value adjusted for moisture content (unitless) (SWAT 2:1.1.11)"""
 
+    # ---- percolation
+    vadose_zone_layer: Optional[LayerData] = None
+    """Datalayer object that represents the vadose zone, arbitrary top and bottom depths, starts with no water"""
+    time_step: float = 24
+    """length of time step over which percolation occurs (hours) """
+
+    def __post_init__(self):
+        if self.soil_layers is None:
+            # sets the soil layers to a default set if user does not provide any
+            self.soil_layers = [LayerData(top_depth=0, bottom_depth=50, nitrate=0.5),
+                                LayerData(top_depth=50, bottom_depth=80, nitrate=1),
+                                LayerData(top_depth=80, bottom_depth=200, nitrate=5)]
+
+        # configures the vadose zone LayerData object based on where the soil profile ends
+        self.vadose_zone_layer = LayerData(top_depth=self.soil_layers[-1].bottom_depth,
+                                           bottom_depth=10000000,    # bottom depth is 10,000 meters by default
+                                           soil_water_concentration=0,
+                                           saturation_point_water_concentration=inf)
+
+    # ---- temperature
+    albedo: float = 0.16
+    """ratio of reflected to solar radiation (unitless)"""
+    # TODO: check above definition with a SME because all I did to get it was google 'soil albedo'
```

Comment:
This is correct, but I'd reword:

```suggestion
    albedo: float = 0.16
    """proportion of solar radiation that is reflected by the soil surface (unitless)"""
```